liquibase-standard/src/main/java/liquibase/precondition/core/ForeignKeyExistsPrecondition.java

```diff
@@ -50,8 +50,12 @@ public void check(Database database, DatabaseChangeLog changeLog, ChangeSet chan
                 example.getForeignKeyTable().setName(getForeignKeyTableName());
             }
             String catalogName = getCatalogName() != null ? getCatalogName() : database.getDefaultCatalogName();
+            String localCatalogName = getCatalogName();
+            if (database.supports(Schema.class) && localCatalogName == null) {
```

Comment:
Should not be `database.supports(Catalog.class)` as we are dealing with `Catalog`  ?

Reply:
This was my idea, and while it solves this issue, I'm not sure if it is correct.  Oracle is backwards (to me), which made me think it needed to be `Schema` instead of `Catalog`.

Reply:
Should this be Oracle-specific?

Reply:
That would be a safer change, yes.